pyOCD/target/cortex_m.py

```diff
@@ -371,31 +374,34 @@ def init(self, initial_setup=True, bus_accessible=True):
                 logging.warning("Unknown AHB IDR: 0x%x" % ahb_idr)
 
         if bus_accessible:
-            self.halt()
+            if self.halt_on_connect:
+                self.halt()
             self.setupFPB()
             self.readCoreType()
             self.checkForFPU()
             self.setupDWT()
-
-            # Build register_list and targetXML
-            self.register_list = []
-            xml_root = Element('target')
-            xml_regs_general = SubElement(xml_root, "feature", name="org.gnu.gdb.arm.m-profile")
-            for reg in self.regs_general:
+            self.buildTargetXML()
+
+    def buildTargetXML(self):
+        # Build register_list and targetXML
+        self.register_list = []
+        xml_root = Element('target')
+        xml_regs_general = SubElement(xml_root, "feature", name="org.gnu.gdb.arm.m-profile")
+        for reg in self.regs_general:
+            self.register_list.append(reg)
+            SubElement(xml_regs_general, 'reg', **reg.gdb_xml_attrib)
+        # Check if target has ARMv7 registers
+        if self.core_type in  (ARM_CortexM3, ARM_CortexM4):
+            for reg in self.regs_system_armv7_only:
+                self.register_list.append(reg)
+                SubElement(xml_regs_general, 'reg',  **reg.gdb_xml_attrib)
+        # Check if target has FPU registers
+        if self.has_fpu:
+            #xml_regs_fpu = SubElement(xml_root, "feature", name="org.gnu.gdb.arm.vfp")
```

Comment:
I accidentally left this commented out line here.  It can be removed, but don't feel obligated to.  The line is:
 #xml_regs_fpu = SubElement(xml_root, "feature", name="org.gnu.gdb.arm.vfp")